source/example1_sync_fifo.rst

```diff
@@ -102,34 +101,31 @@ below is an implementation for sync fifo which:
     //=================================================================
         always_ff @(posedge clk or negedge rst_n) begin
             if (~rst_n) begin
-                wr_ptr      <= {AWIDTH+1{1'b0}};
-                rd_ptr      <= {AWIDTH+1{1'b0}};
-                // dummy_wire1 <= 1'b0;
-                // dummy_wire1 <= 1'b0;
+                wr_ptr      <= {AWIDTH{1'b0}};
+                rd_ptr      <= {AWIDTH{1'b0}};
             end
             else begin
                 if (rd_en) begin
                     if (rd_ptr < DEPTH-1)
                         rd_ptr <= type(rd_ptr)'(rd_ptr + 1'b1);
                     else 
-                        rd_ptr <= {AWIDTH+2{1'b0}};
+                        rd_ptr <= {AWIDTH{1'b0}};
                 end
                 if (wr_en) begin
                     if (wr_ptr < DEPTH-1)
```

Comment:
it seems that it shoulb be "< DEPTH" or "<= (DEPTH-1)" ?

Reply:
lets say DEPTH = 8, thus, the bit length of rd_ptr should be 3 which represent 0~7 index.
if we use < 8 or <= 7 will push rd_ptr to 8(4'b1000) eventually. 
in that case we dont need logic from line 111~112. 
otherwise, we keep <7, this will push rd_prt to 7(3'b111) eventually, and overflow will not have chance to happen.
in that case, line 111~112 will be responseble for turning ptr back to 0 position.
I think both them are correct.

Reply:
lets say DEPTH = 8, thus, the bit length of rd_ptr should be 3 which represent 0-7 index.
if we use < 8 or <= 7 will push rd_ptr to 8(4'b1000) eventually.
in that case we dont need logic from line 111-112.
otherwise, we keep <7, this will push rd_prt to 7(3'b111) eventually, and overflow will not have chance to happen.
in that case, line 111~112 will be responseble for turning ptr back to 0 position.
I think both them are correct.

Reply:
but, it seems like your recommendation will reduce a comparator and mux. I will change this later